validate/src/request.rs

```diff
@@ -903,17 +960,104 @@ pub fn template_name(name: impl AsRef<str>) -> Result<(), ValidationError> {
 /// [`Username`]: ValidationErrorType::Username
 /// [this documentation entry]: https://discord.com/developers/docs/resources/user#usernames-and-nicknames
 pub fn username(value: impl AsRef<str>) -> Result<(), ValidationError> {
-    let len = value.as_ref().chars().count();
+    let value = value.as_ref();
+    let len = value.chars().count();
+
+    let invalid_len = if (USERNAME_LIMIT_MIN..=USERNAME_LIMIT_MAX).contains(&len) {
+        None
+    } else {
+        Some(len)
+    };
+
+    let invalid_substring = self::invalid_substring(value);
 
-    if (USERNAME_LIMIT_MIN..=USERNAME_LIMIT_MAX).contains(&len) {
+    if invalid_len.is_none() && invalid_substring.is_none() {
         Ok(())
     } else {
         Err(ValidationError {
-            kind: ValidationErrorType::Username { len },
+            kind: ValidationErrorType::Username {
+                len: invalid_len,
+                substring: invalid_substring,
+            },
         })
     }
 }
 
+/// Ensure that a webhook is correct.
+///
+/// The length must be at least [`WEBHOOK_USERNAME_LIMIT_MIN`] and at most
+/// [`WEBHOOK_USERNAME_LIMIT_MAX`]. It must also be free of certain substrings.
+/// This is based on [this documentation entry].
+///
+/// # Errors
+///
+/// Returns an error of type [`WebhookUsername`] if the length is invalid.
+///
+/// [`WebhookUsername`]: ValidationErrorType::WebhookUsername
+/// [this documentation entry]: https://discord.com/developers/docs/resources/webhook#create-webhook
+pub fn webhook_username(value: impl AsRef<str>) -> Result<(), ValidationError> {
+    let value = value.as_ref();
+    let len = value.chars().count();
+
+    let invalid_len = if (WEBHOOK_USERNAME_LIMIT_MIN..=WEBHOOK_USERNAME_LIMIT_MAX).contains(&len) {
+        None
+    } else {
+        Some(len)
+    };
+
+    let invalid_substring = if value.to_ascii_lowercase() == CLYDE {
+        Some(CLYDE)
+    } else {
+        None
+    };
```

Comment:
Through the power of `bool::then`:
```suggestion
    let len_range = (WEBHOOK_USERNAME_LIMIT_MIN..=WEBHOOK_USERNAME_LIMIT_MAX);
    let invalid_len = (!len_range.contains(&len)).then(|| len);
  
    let invalid_substring = (value.to_ascii_lowercase() == CLYDE).then(|| CLYDE);
```